File: pueblo-golf-tournament-api/Dto/Outgoing/ReceiveLeaderboardUpdateResponseDto.cs
namespace pueblo_golf_tournament_api.Dto.Outgoing
{
    public class ReceiveLeaderboardUpdateResponseDto
    {
        public List<DivisionLeaderBoard> Data { get; set; }
    }

    public class DivisionLeaderBoard { 
        public DivisionDto Division { get; set; }
        public List<LeaderboardDto> LeaderBoard { get; set; } = new List<LeaderboardDto>();
    }
}

File: pueblo_golf_tournament_mobile/lib/dto/response/receive-leader-board-update-response-dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pueblo_golf_tournament_mobile/base/base-response.dart';
import 'package:pueblo_golf_tournament_mobile/dto/model/leader-board-dto.dart';

import '../model/division-dto.dart';
import 'lookup-leader-board-response-dto.dart';

part 'receive-leader-board-update-response-dto.g.dart';

@JsonSerializable()
class ReceiveLeaderBoardUpdateResponseDto {
  final List<DivisionLeaderBoard> data;

  ReceiveLeaderBoardUpdateResponseDto({required this.data});

  factory ReceiveLeaderBoardUpdateResponseDto.fromJson(
          Map<String, dynamic> json) =>
      _$ReceiveLeaderBoardUpdateResponseDtoFromJson(json);

  Map<String, dynamic> toJson() =>
      _$ReceiveLeaderBoardUpdateResponseDtoToJson(this);
}
